client/src/periodUtils.js

```diff
@@ -0,0 +1,152 @@
+import moment from "moment"
+import PropTypes from "prop-types"
+import React from "react"
+
+const now = moment()
+
+export const RECURRENCE_TYPE = {
+  ONCE: "once",
+  DAILY: "daily",
+  WEEKLY: "weekly",
+  BIWEEKLY: "biweekly",
+  SEMIMONTHLY: "semimonthly",
+  MONTHLY: "monthly",
+  QUARTERLY: "quarterly",
+  SEMIANNUALY: "semiannualy"
+}
+
+export const PERIOD_FACTORIES = {
+  [RECURRENCE_TYPE.DAILY]: (date, offset) => ({
+    start: date.clone().subtract(offset, "days").startOf("day"),
+    end: date.clone().subtract(offset, "days").endOf("day")
+  }),
+  [RECURRENCE_TYPE.WEEKLY]: (date, offset) => ({
+    start: date.clone().subtract(offset, "weeks").startOf("week"),
+    end: date.clone().subtract(offset, "weeks").endOf("week")
+  }),
+  // FIXME: biweekly should be each 2 weeks from the beginning of the year
```

Comment:
I don't know what this means. When should a biweekly period start, and when should it end? (also, be aware of the ambiguity of the word; it could also mean 'twice a week' which is four times as often :wink: ; perhaps consider using "fortnightly")

Reply:
Me neither :-)
I've added an issue for it a while ago: https://github.com/NCI-Agency/anet/issues/3068
For now I've changed the comment to "FIXME: biweekly calculation should be changed, first agree on what it means".